src/app/courses/course-plate/course-plate.component.ts

```diff
@@ -1,11 +1,12 @@
-import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
 import { CourseModel } from '../course.model';
 import { Guid } from 'guid-typescript';
 import { faStar } from '@fortawesome/free-solid-svg-icons';
 
 @Component({
   selector: 'app-course-plate',
   templateUrl: './course-plate.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
Also you may put this strategy to whole container (I mean course list component)
But in this case you should provide new array object each time, when new data is arrived.